pkg/xds/topology/outbound.go

```diff
@@ -154,6 +154,36 @@ func fillDataplaneOutbounds(
 	}
 }
 
+func BuildCrossMeshEndpointMap(
+	gateways *core_mesh.MeshGatewayResourceList,
+	dataplanes *core_mesh.DataplaneResourceList,
+) core_xds.EndpointMap {
+	outbound := core_xds.EndpointMap{}
+	for _, dataplane := range dataplanes.Items {
+		gateway := SelectGateway(gateways, dataplane.Spec.Matches)
+		if !dataplane.Spec.IsBuiltinGateway() {
+			continue
+		}
+		dpSpec := dataplane.Spec
+		dpNetworking := dpSpec.GetNetworking()
+
+		dpGateway := dpNetworking.GetGateway()
+		dpTags := dpGateway.GetTags()
+		serviceName := dpTags[mesh_proto.ServiceTag]
+
+		for _, listener := range gateway.Spec.Conf.Listeners {
+			outbound[serviceName] = append(outbound[serviceName], core_xds.Endpoint{
+				Target: dpNetworking.GetAddress(),
+				Port:   listener.GetPort(),
+				Tags:   mesh_proto.Merge(dpTags, gateway.Spec.GetTags(), listener.GetTags()),
+				Weight: 1,
+				// TODO Locality: ,
```

Comment:
still missing, if not for this PR, please create an issue so we don't forget to implement it

Reply:
This one I lost track of, it's an early TODO, I'll take another look

Reply:
I set it now, I'll reexamine when I add multizone